ruby/src/IceRuby/Operation.cpp

```diff
@@ -278,7 +275,7 @@ IceRuby::OperationI::OperationI(VALUE name, VALUE mode, VALUE sendMode, VALUE am
 }
 
 VALUE
-IceRuby::OperationI::invoke(const Ice::ObjectPrx& proxy, VALUE args, VALUE hctx)
+IceRuby::OperationI::invoke(const shared_ptr<Ice::ObjectPrx>& proxy, VALUE args, VALUE hctx)
```

Comment:
See https://github.com/zeroc-ice/ice/issues/1704

It would be nice if we could keep the old signature (`const Ice::ObjectPrx& proxy`).

However, our C++11 proxies currenly heavily rely on shared_ptr in their implementation. For example:
https://github.com/zeroc-ice/ice/blob/9af8a3758e88813eb840f01aed7800aefa1a5ef7/cpp/include/Ice/Proxy.h#L592

So we can't do that yet.